src/libsystemd/sd-journal/journal-file.c

```diff
@@ -918,6 +947,26 @@ static int journal_file_check_object(JournalFile *f, uint64_t offset, Object *o)
                                                le64toh(o->tag.epoch), offset);
 
                 break;
+
+        case OBJECT_TRIE_NODE:
```

Comment:
The check for OBJECT_ENTRY_ARRAY further up doesn't take compact headers into account (same in journal-verify)